flaml/default/suggest.py

```diff
@@ -178,9 +178,10 @@ def suggest_hyperparams(task, X, y, estimator_or_predictor, location=None):
     """
     config = suggest_config(task, X, y, estimator_or_predictor, location=location, k=1)[0]
     estimator = config["class"]
-    model_class = get_estimator_class(task, estimator)
+    task = task_factory(task)
+    model_class = task.estimator_class_from_str(estimator)
     hyperparams = config["hyperparameters"]
-    model = model_class(task=task, **hyperparams)
+    model = model_class(task=task.name, **hyperparams)

```

Comment:
Is there a reason for this change? It seems to make it less general.

Reply:
As discussed, previously `task` was a string not an object, so this code is still getting the string that it expects.